javascript-loops/main.js

```diff
@@ -0,0 +1,73 @@
+/* exported getNumbersToTen,
+            getEvenNumbersToTwenty,
+            repeatWord,
+            logEachCharacter,
+            doubleAll,
+            getKeys,
+            getValues
+ */
+function getNumbersToTen() {
+  var numbers = [];
+  var currentNumber = 1;
+
+  for (;currentNumber <= 10; currentNumber++) {
+    numbers[currentNumber - 1] = currentNumber;
+  }
+  return (numbers);
+}
+console.log(getNumbersToTen());
+
+function getEvenNumbersToTwenty() {
+  var evenNumbers = [];
+  var currentNumber = 2;
+
+  for (;currentNumber <= 10; currentNumber += 2) {
+    evenNumbers[currentNumber / 2 - 1] = currentNumber;
+  }
+  return (evenNumbers);
+}
+console.log(getEvenNumbersToTwenty());
+
+function repeatWord(word, times) {
+  var count = 1;
+  var repeated = '';
+
+  for (;count <= times; count++) {
+    repeated += word;
+  }
+  return (repeated);
+}
+console.log(repeatWord('derp', 10));
+
+function logEachCharacter(string) {
+  for (var i = 0; i < string.length; i++) {
+    console.log(string.charAt(i));
+  }
+}
+logEachCharacter('funkadelic');
+
+function doubleAll(array) {
+  for (var i = 0; i < array.length; i++) {
+    array[i] *= 2;
```

Comment:
This will mutate the original array, you should create a new array and push the doubles into the new array and return that.